nixos/modules/config/xdg/portal.nix

```diff
@@ -0,0 +1,38 @@
+{ config, pkgs ,lib ,... }:
+with lib;
+{
+  options.xdg.portal={
+    enable =
+      mkEnableOption "(xdg desktop integration)[https://github.com/flatpak/xdg-desktop-portal]"//{
+        default = true;
+      };
+
+    extraPortals = mkOption {
+      type = types.listOf types.package;
+      default = [];
+      description = ''
+        List of additional portals to add to path. Portals allow interaction
+        with system, like choosing files or taking screenshots. At minimum,
+        a desktop portal implementation should be listed. GNOME and KDE already
+        adds <package>xdg-desktop-portal-gtk</package>; and
+        <package>xdg-desktop-portal-kde</package> respectively. On other desktop
+        environments you probably want to add them yourself.
+      '';
+    };
+  };
+
+  config =
+    let
+      cfg=config.xdg.portal;
+      portals=[pkgs.xdg-desktop-portal] ++ cfg.extraPortals;
```

Comment:
`xdg-desktop-portal` itself isn't strictly speaking a *portal*, is it? Although we do certainly want to add it to systemd packages and dbus packages.